cmd/eksctl-anywhere/cmd/flags.go

```diff
@@ -0,0 +1,42 @@
+package cmd
+
+import (
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+	"github.com/spf13/viper"
+)
+
+func addClusterFlags(cc *cobra.Command, clusterOpt *clusterOptions, override ...string) {
+	overriddenFlags := make(map[string]struct{})
+	for _, flag := range override {
+		overriddenFlags[flag] = struct{}{}
+	}
+
+	if _, ok := overriddenFlags["filename"]; !ok {
+		cc.Flags().StringVarP(&clusterOpt.fileName, "filename", "f", "", "Filename that contains EKS-A cluster configuration")
+	}
+	if _, ok := overriddenFlags["bundles-override"]; !ok {
+		cc.Flags().StringVar(&clusterOpt.bundlesOverride, "bundles-override", "", "Override default Bundles manifest (not recommended)")
+	}
+	if _, ok := overriddenFlags["kubeconfig"]; !ok {
+		cc.Flags().StringVar(&clusterOpt.managementKubeconfig, "kubeconfig", "", "Management cluster kubeconfig file")
+	}
+	if _, ok := overriddenFlags["install-packages"]; !ok {
+		cc.Flags().StringVar(&clusterOpt.installPackages, "install-packages", "", "Location of curated packages configuration files to install to the cluster")
+	}
+	if _, ok := overriddenFlags["force-cleanup"]; !ok {
+		cc.Flags().BoolVar(&clusterOpt.forceCleanup, "force-cleanup", false, "Force deletion of previously created bootstrap cluster")
+	}
+}
+
+func bindAllFlags(cmd *cobra.Command, args []string) error {
+	var err error
+
+	cmd.Flags().VisitAll(func(flag *pflag.Flag) {
+		if err != nil {
+			return
+		}
+		err = viper.BindPFlag(flag.Name, flag)
```

Comment:
this would override err if a the first flag fails but the second succeeds

Reply:
I grabbed this logic from Chris's comment on the preflight POC: https://github.com/aws/eks-anywhere/pull/2858#discussion_r938733162

From my understanding this initializes err in scope of the bindAllFlags parent function and stops binding flags if that err value is not nil.

Reply:
It won't override. After the first error is observed it returns on l.37 for all subsequent visits and finally returns the original error on l.41.